### ISSUE TITLE
Add daily live-install backup driven by the bridge daemon

### ISSUE THREAD

**Issue by seanssoh** [SYRS-AI/agent-bridge-public#66]
## Symptom

Live install (agents/, agent-roster.local.sh, state/, auth tokens, operator customizations) has no built-in backup. Any host loss or accidental rm deletes all operator work. upgrade/ rollback only covers core script regressions, not data protection.

## Why this looks upstream

Operators currently have to roll their own tar+cron or git-init their live install. Agent Bridge's upgrade mechanism already demonstrates the 'preserve operator state' pattern and has a 'backups/' dir for upgrade rollbacks, so extending the daemon with a scheduled daily snapshot reuses existing primitives. Missing feature for a multi-agent durability-oriented product.

## Reproduction

```text
# Feature request: daily live-install backup daemon

## Motivation

Agent Bridge `upgrade` preserves operator state (`agents/<agent>/`, `agent-roster.local.sh`, `state/`, `shared/`, etc.) across core upgrades. But **nothing protects the live install against**:

- host disk loss / instance termination
- accidental `rm -rf` on agent home or roster file
- credential file corruption (Discord/Telegram/Teams tokens, auth.json under channel state dirs)
- operator customizations (CLAUDE.md edits, per-agent `.mcp.json`, project-specific skills)

Right now, operators have to roll their own tar+cron or git-init their live install. For a product that explicitly positions itself around multi-agent durability, missing built-in backup is a surprising gap.

## Proposal

Add a built-in daily backup driven by the existing daemon. No new long-running process. No external deps.

### Defaults
- **Enabled by default** (operator can disable with `BRIDGE_DAILY_BACKUP_ENABLED=0`)
- **Schedule**: once per day in early morning local time (default `04:00`, configurable `BRIDGE_DAILY_BACKUP_HOUR=4`, `BRIDGE_DAILY_BACKUP_MINUTE=0`)
- **Output**: `${BRIDGE_DAILY_BACKUP_DIR:-$BRIDGE_HOME/backups/daily}/agent-bridge-<YYYY-MM-DD>.tgz`
- **Retention**: keep last N days (default 30, `BRIDGE_DAILY_BACKUP_RETAIN_DAYS=30`); older files deleted
- **Atomicity**: tar to `<name>.tmp.<pid>` then `mv` on success

### What's included
Entire `$BRIDGE_HOME` except:
- `logs/**` — churny, too large
- `backups/**` — avoid recursion
- `state/daemon.log` — churny (but we keep `state/` for roster/session metadata)
- `*.pid`, unix sockets, partial files
- anything matching an operator-supplied `BRIDGE_DAILY_BACKUP_EXCLUDE_GLOBS` (comma-sep csv)

### What's NOT done automatically
- No offsite sync (S3/git). Operator opts in separately — e.g., hook a post-backup command via `BRIDGE_DAILY_BACKUP_POST_HOOK` so they can push to S3/rsync/etc.
- No encryption. Local filesystem is trusted. If operator wants, they use `BRIDGE_DAILY_BACKUP_POST_HOOK` to run `gpg --encrypt`.
- No secret redaction. This is a local backup of a local install; secrets roundtrip.

### CLI surface
```
agent-bridge backup now [--dry-run] [--label <text>]       # manual on-demand
agent-bridge backup list [--json]                           # list retained archives
agent-bridge backup restore <archive> [--target <dir>]     # restore into an empty or specific dir
agent-bridge backup prune [--retain <N>]                    # honor retention manually
```

### Daemon wiring
Reuse the existing scheduler pattern that fires plugin MCP liveness and memory daily refresh. Pseudocode near `process_memory_daily_refresh_requests` in `bridge-daemon.sh`:

```bash
process_daily_backup() {
  [[ "${BRIDGE_DAILY_BACKUP_ENABLED:-1}" == "1" ]] || return 0
  local now_ts hour minute marker
  hour="$(date +%H)" ; minute="$(date +%M)"
  [[ "$hour" == "$(printf '%02d' "${BRIDGE_DAILY_BACKUP_HOUR:-4}")" ]] || return 0
  [[ "$minute" =~ ^${BRIDGE_DAILY_BACKUP_MINUTE:-0}[0-9]?$ ]] || return 0   # ~10m window
  marker="$BRIDGE_STATE_DIR/daily-backup/$(date +%Y-%m-%d).done"
  [[ -f "$marker" ]] && return 0
  bridge_daily_backup_run && mkdir -p "$(dirname "$marker")" && touch "$marker"
}
```

The actual tar step lives in a new `bridge-backup.sh` keyed off `$BRIDGE_HOME`, with exclusion list and atomic rename.

### Why upstream

- Disaster recovery is a cross-cutting operator concern, not per-install.
- Single implementation avoids every operator reinventing tar+cron with subtly different exclusions.
- Existing `backups/` directory semantics already exist for upgrade rollback — reusing the name with a `daily/` subdir keeps mental model consistent.

### Acceptance criteria
1. Fresh `agent-bridge upgrade` to a version containing this change + 24h wait → `backups/daily/agent-bridge-YYYY-MM-DD.tgz` exists for today.
2. File contains `agent-roster.local.sh`, `agents/<agent>/` (incl. operator customizations), `shared/`, `state/` (minus daemon.log).
3. File does NOT contain `logs/`, `backups/`, `.pid`, sockets.
4. Older backups are pruned per retention.
5. `agent-bridge backup restore <archive> --target /tmp/test` cleanly extracts with operator files intact.
6. `BRIDGE_DAILY_BACKUP_ENABLED=0` fully disables the scheduled run; manual `agent-bridge backup now` still works.

### Open questions for upstream review
- Should the default destination be inside `$BRIDGE_HOME/backups/daily/` or an external path (e.g. `~/agent-bridge-backups/`)? Default inside = self-contained but vulnerable to same-host disk loss. External = safer but requires path decision per-host.
- Retention: count-based (last 30) vs age-based (older than 30 days) vs hybrid. Current proposal is count-based; this is simpler.
- Should `state/history/` and `state/timestamps/` be excluded (regenerable) or kept (supports full-state restore)? Proposal keeps them.

## Impact if not addressed
Operators run `rm -rf` at their peril. Any host loss = full manual reconstruction of:
- roster (agent list, launch cmds, channel metadata)
- auth tokens for every channel on every agent
- operator CLAUDE.md customizations
- memory wiki content
- shared knowledge base

All of which are trivially zippable and restorable if backup existed.
```

## Environment

- Agent Bridge version: 0.2.16
- Agent Bridge source: /home/ec2-user/.agent-bridge
- OS: Linux ip-10-242-2-60.ap-northeast-2.compute.internal 6.1.94-99.176.amzn2023.x86_64 #1 SMP PREEMPT_DYNAMIC Tue Jun 18 14:57:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
- Shell: /bin/bash

## Consent

This draft must not be filed until the user explicitly approves.
